### PR TITLE
docs(server): update server guides to 1.0

### PR DESCRIPTION
**Pull request by oddgrd**
This PR updates the server guides to 1.0, partially addressing https://github.com/hyperium/hyper/issues/3068. I realized when starting this that the examples currently link to the master branch of hyper, so I'll create a PR for https://github.com/hyperium/hyper/issues/3069 as well soon if it isn't picked up. I'm also wondering if we should merge these PRs to master, or keep them in a separate dev branch until 1.0 is released?

While updating these guides, I'll also try to make a note of API docs that feel lacking. 

And of course, any feedback is very welcome. :slightly_smiling_face: 

Closes https://github.com/hyperium/hyper/issues/3083
Closes https://github.com/hyperium/hyper/issues/3069

### CODE REVIEW

**Comment by seanmonstar** [comment]
Oh, seeing no CI results, I'm reminded this repo still uses the now-defunct travis-ci. We'll need to convert it over to GH Actions.

**Comment by seanmonstar** [comment]
Alright, got CI working. I think what we'll notice now is that since _some_ of the guides use hyper 0.14, and the new ones will be using 1.0, CI isn't setup to handle that.

Soooo, I'm thinking next step is having a 0.14 and 1.0 section, for now.

**Comment by oddgrd** [comment]
Hi, thanks for sorting out the CI! So are you thinking something like `guides/0.14/...` and `guides/1.0/...`? I'll update the links as well then, since they are linked to master and not 0.14.

**Comment by seanmonstar** [comment]
> guides/0.14/... and guides/1.0/...

Yes, I believe so. I might go with `guides/0.14/...` and `guides/1/...`. The existing guides should probably redirect to the `0.14` ones, for now. We'll change them to `1` eventually.

**Comment by oddgrd** [comment]
I managed to split up the guides now, but the solution may be a bit hacky as I'm not familiar with jekyll. Now this PR has expanded a bit much in scope. So to get this over the line I will cut out `graceful_shutdown` from the 1.0 guides for now and I'll add it back in a follow-up PR. 

Following that, what remains for this PR is to make choosing 1.0 or 0.14 guides ergonomic. I'm thinking `get started` goes to 1.0 by default, and we can have tabs in the nav bar for switching versions. And then we can merge this (pending review of course).

PS. I also edited the CI script so it will test both versions. The script isn't very kind on the eyes, but I may be able to clean it up a bit tomorrow.

File: _config.yml [hunk 1]
```diff
@@ -6,13 +6,24 @@ url: "https://hyper.rs" # the base hostname & protocol for your site
...
+      path: "_stable/"
+      type: "guides"
+    values:
+      layout: "guide"
+  
 
 collections:
-  guides:
+  legacy:
+    permalink: /:collection/:path/
```
Comment (seanmonstar) [inline _config.yml:24]: Is it not possible to set the URL here? `/guides/0.14/:path/` and `/guides/1/:path`?

Comment (oddgrd) [reply _config.yml]: It was indeed, thanks!

**Comment by seanmonstar** [comment]
Awesome! I think one last thing, and I think we're set: do we need to use [redirects](https://github.com/jekyll/jekyll-redirect-from) so that links to the existing guides will still work after?

**Comment by oddgrd** [comment]
Sure! I'll add that later today. I have also "defaulted" to the 1.0 guides (Get Started leads to 1.0, front page is 1.0), but with links to 0.14 guides. The 1.0 guides are quite sparse still, so if you'd rather we default to 0.14 for now I can quickly do that as well.

Either way I do plan to add more updated guides this week.

**Review by seanmonstar — APPROVED** [review]
Phenomenal work! 🥳

**Pull request merged**